File: gnsals0904/level1/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int paintedEnd = 0;

        for (int posi : section) {
            if (posi <= paintedEnd) {
                continue;
            }

            answer++;

            int start = Math.min(posi, n - m + 1);
            paintedEnd = start + m - 1;
        }

        return answer;
    }
}

File: yujin-level0/Week64/덧칠하기.js
function solution(n, m, section) {
  let count = 0;
  let paintedUntil = 0; // 롤러가 덮은 마지막 구역 번호

  for (let s of section) {
    // 현재 칠해야 할 구역이 아직 칠해지지 않은 위치라면
    if (s > paintedUntil) {
      count++;
      paintedUntil = s + m - 1;
    }
  }

  return count;
}
